__tests__/api.ts

```diff
@@ -64,6 +65,47 @@ test('Calls to API get a signature', async () => {
   })
 })
 
-test('header `Access-Control-Allow-Origin` is fixed', () => {
-  expect(response.headers.get('Access-Control-Allow-Origin')).toBe('origin')
+async function fetchWithOriginHeader(origin: string) {
+  return await env.fetch(url, {
+    ...requestPayload,
+    headers: { ...requestPayload.headers, Origin: origin },
+  })
+}
+
+test("header `Access-Control-Allow-Origin` is set to Serlo's domain or subdomains", async () => {
+  const domainOrigin = `https://${env.getDomain()}`
+
+  const responseWithoutOriginHeader = response
+
+  expect(
+    responseWithoutOriginHeader.headers.get('Access-Control-Allow-Origin')
+  ).toBe(domainOrigin)
+
+  const responseWithRightDomain = await fetchWithOriginHeader(domainOrigin)
+
+  expect(
+    responseWithRightDomain.headers.get('Access-Control-Allow-Origin')
+  ).toBe(domainOrigin)
+
+  const subdomainOrigin = `https://de.${env.getDomain()}`
+
+  const responseWithRightSubdomain = await fetchWithOriginHeader(
+    subdomainOrigin
+  )
+
+  expect(
+    responseWithRightSubdomain.headers.get('Access-Control-Allow-Origin')
+  ).toBe(subdomainOrigin)
+
+  const responseWithWrongOrigin = await fetchWithOriginHeader(
+    `https://verybad-${env.getDomain()}`
+  )
+
+  expect(
+    responseWithWrongOrigin.headers.get('Access-Control-Allow-Origin')
+  ).not.toBe(`https://verybad-${env.getDomain()}`)
+})
```

Comment:
This should be four different tests. Make sure to always use the AAA-pattern here ( https://medium.com/xebia-engineering/unit-testing-aaa-pattern-ab1c08737d53 )